src/datastore/postgres-store.ts

```diff
@@ -2936,6 +2937,48 @@ export class PgDataStore extends (EventEmitter as { new (): DataStoreEventEmitte
     });
   }
 
+  async getAddressNFTEvent(args: {
+    stxAddress: string;
+    limit: number;
+    offset: number;
+  }): Promise<{ results: AddressNftEventIdentifier[]; total: number }> {
+    return this.query(async client => {
+      const result = await client.query<AddressNftEventIdentifier & { count: string }>(
+        `
+        WITH lastAddressTransfers AS (
```

Comment:
It looks like this query can return incorrect NFT ownership if an NFT's ownership is transferred more than once per block, for example in different transactions within a block or if a single transaction moved the NFT multiple times. 

To fix this the query needs to get the latest by `block_height DESC, tx_index DESC, event_index DESC` as opposed to only `block_height`. 

The multicolumn filtering can be done using a `DISTINCT ON(...) ... GROUP BY ...`, for example:

```sql
WITH address_transfers AS (
  SELECT asset_identifier, value, sender, recipient, block_height, tx_index, event_index
  FROM nft_events
  WHERE canonical = true AND recipient = $1
),
last_nft_transfers AS (
  SELECT DISTINCT ON(asset_identifier, value) asset_identifier, value
  FROM nft_events
  WHERE canonical = true
  ORDER BY asset_identifier, value, block_height DESC, tx_index DESC, event_index DESC
)
SELECT sender, recipient, asset_identifier, value, COUNT(*) OVER() AS count
FROM address_transfers INNER JOIN last_nft_transfers USING (asset_identifier, value)
ORDER BY block_height DESC, tx_index DESC, event_index DESC
LIMIT $2 OFFSET $3
```